src/Batch/BatchApiV2.php

```diff
@@ -63,6 +63,21 @@ public function createBatch(
         if (count($fileUris) === 0) {
             throw new \UnexpectedValueException('FileUris cannot be empty.');
         }
+        if (!function_exists('array_is_list')) {
+            function array_is_list(array $array): bool
+            {
+                $pointer = -1;
+                foreach ($array as $key => $_) {
+                    if ($key !== $pointer) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+        }
```

Comment:
I don't get it. According to php doc `An array is considered a list if its keys consist of consecutive numbers from 0 to count($array)-1.` You pointer never gets incremented to check key against it. Your function will always return false if array doesn't;t start with element with key `-1`, now? Let's write tests for this function.

Reply:
Added, great catch, thanks. The tests were present, but I'm using php 8.2, so the polyfill code was not called